### DISCUSSION TITLE
Naming relation column in an explicit m-n relation in the same manner as it is named in an implicit m-n relation

### DISCUSSION THREAD

**Discussion by Nyloth9** [prisma/prisma#16838] [Q&A] [answered]
Hello, I was wondering if it is possible and how to achieve it; if I can have the same naming convention in an explicit m-n relation as it is in an implicit m-n relation. Here is what I mean:

When i make an implicit relation between Person and Category like this:
<img width="543" alt="Screenshot 2022-12-15 at 13 47 44" src="https://user-images.githubusercontent.com/96616657/207863620-70f453f6-fe7d-4079-8fc0-0be921856001.png">

- I get a relationship in my database that looks like this (categories on person are shown as "Categories"): 
<img width="1395" alt="Screenshot 2022-12-15 at 13 57 50" src="https://user-images.githubusercontent.com/96616657/207865037-583767bd-2601-4b4d-b199-9869c011406c.png">

and in the same manner, relation column for "person" in "categories" is shown as "Person".

---

However when i make an explicit m-n relation like this:
<img width="557" alt="Screenshot 2022-12-15 at 14 01 37" src="https://user-images.githubusercontent.com/96616657/207865640-865eb5f0-411e-406d-bb3c-55ff79955441.png">

- I get a relationship in my database that looks like this (categories on person are shown as "CategoriesOnPersons"):
<img width="1412" alt="Screenshot 2022-12-15 at 13 54 06" src="https://user-images.githubusercontent.com/96616657/207865426-ce1a92b1-a609-4c32-868b-9e9d23099646.png">

and in the same manner, relation column for "person" in "categories" is shown as "CategoriesOnPersons".

---

I like very much how the implicit relation displays these related columns, and I would like for when I make explicit m-n relation that in the table "Persons" the "categories" column shows "Category" and not "CategoriesOnPersons", same as it is in an implicit m-n relationship. I have tried to play around a bit with mapping and naming the relation, and none of those worked, so I was wondering, how can this be achieved in an explicit m-n relation. 

Thank you

**Comment by nurul3101** [comment, marked as answer]
Hey @Nyloth9 👋 

I don't think what you are trying to achieve is possible yet. You can rename the Connecting table in database to a different name using [@@map](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#map-1) attribute, but the name would appear similar. So, Persons and Categories table would show the same column name unlike naming in implicit many-to-many.

**Reply by Nyloth9** [reply]
Ah thats unfortunate. Thank you for your reply

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
